### ISSUE TITLE
Images in jsx don't get detected

### ISSUE THREAD

**Issue by DeMoorJasper** [parcel-bundler/parcel#379]
🐛 bug report

When using JSX my images don't get resolved.

```jsx
<img src="/dist/assets/logo.png" alt="logo" title="logo" class='logo' />
```
### 🎛 Configuration (.babelrc, package.json, cli command)

.babelrc
```json
{
  "plugins": [
    [
      "transform-react-jsx",
      {
        "pragma": "h"
      }
    ]
  ]
}
```
### 🤔 Expected Behavior

It should work

### 😯 Current Behavior

It doesn't work, it just keeps the path as specified in jsx

### 🔦 Context

Preact application

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.3.0
| Node             | 9.2.0
| npm/Yarn         | yarn 1.3.2
| Operating System | Mac OS X High Sierra

**Comment by kingdaro** [comment]
_Is_ this a bug? JSX is just JavaScript, so interpreting an image path in JSX would be the same as interpreting any old string. The solution here would be to import/require it.
```jsx
<img src={require("/dist/assets/logo.png")} alt="logo" title="logo" class='logo' />
```

**Comment by shawwn** [comment]
That's correct.

However, it may be worth special casing this. It doesn't matter to users
how correct it is -- having to write require("...") for an image tag is
very counterintuitive, even if it makes sense.

We ought to do the same logic for JSX that we do with posthtml.

**Comment by DeMoorJasper** [comment]
@kingdaro thanks for the workaround but i think eventually this should work in both ways

**Comment by devongovett** [comment]
This will be hard to do correctly since it would only work for images where the `src` property is a string literal. JSX can have variables in it, or components which wrap an `img` tag and pass `src` through as a prop. HTML cannot so it's easy to statically analyze.

We can statically analyze `require` calls though, the result of which can be assigned to a variable and passed around.

**Comment by hufei1993** [comment]
it does not work when I use <img className="App-Logo" src={require('./test.jpg')} alt="React Logo" />

**Comment by MariusAlch** [comment]
```
  const cat = require('./cat.jpg');
...
  render() {
    return (
        <img src={`/dist${cat}`} alt="cat"/>
    );
  }
```

This works

**Comment by shawwn** [comment]
Hmmm, this seems like a bug. Thanks for the report.

**Comment by thejameskyle** [comment]
I don't think we should even attempt to support `<img src="...">`. As soon as it is dynamic it can no longer be supported. It's better to be consistent and use `require()` and have that return the public path.

**Comment by devongovett** [comment]
Agreed. Closing this issue.

The issue with requiring urls to files should be fixed now in v1.3.1.

**Comment by shawwn** [comment]
Agreed, but they're saying the path returned by require is wrong. (It doesn't include the leading /dist/)

**Comment by devongovett** [comment]
yep, but that issue should be fixed in v1.3.1.
